### PR TITLE
70 add support for using new v2 visualization endpoints

### PR DESCRIPTION
**Pull request by sandeep-ps**
Steps for reviewing this PR:

1. Run the Clowder stack (all services, frontend, and backend)

2. Install and run the example image preview extractor (sample-extractors/image-preview-v2) by following the instructions in its README. The Docker instructions won't work now as it uses the PyClowder image as the base. I have changed the requirements.txt to use `git+https://github.com/clowder-framework/pyclowder.git@70-add-support-for-using-new-v2-visualization-endpoints`. It'll be changed to `pyclowder` before merging. 

3. Upload an image file to the Clowder instance, and you should see the updates in the extractor logs. Once Clowder's visualization download URL is updated, the new visualization (image preview) should appear in the user interface.

Note: The following environment variables need to be set when running the extractor:
```
CLOWDER_VERSION=2
IMAGE_BINARY="<Output of `which convert` command>"
```
ImageMagick can be installed from here: https://imagemagick.org/script/download.php
This extractor uses the `convert` tool from ImageMagick. You can find the installed binary file path of `convert` using the `which convert` command and use that in the IMAGE_BINARY environment variable above.

### CODE REVIEW

File: pyclowder/api/v1/files.py [hunk 1]
```diff
@@ -202,7 +205,7 @@ def upload_metadata(connector, client, fileid, metadata):
 
 
 # pylint: disable=too-many-arguments
-def upload_preview(connector, client, fileid, previewfile, previewmetadata=None, preview_mimetype=None):
+def upload_preview(connector, client, fileid, previewfile, previewmetadata=None, preview_mimetype=None, **kwargs):
```
Comment (longshuicy) [inline pyclowder/api/v1/files.py:208]: good idea of using **kwargs for backward compatibility

File: pyclowder/api/v2/files.py [hunk 2]
```diff
@@ -173,36 +171,69 @@ def upload_preview(connector, client, fileid, previewfile, previewmetadata=None,
...
-    with open(previewfile, 'rb') as filebytes:
-        # If a custom preview file MIME type is provided, use it to generate the preview file object.
-        if preview_mimetype is not None:
-            result = connector.post(url, files={"File": (os.path.basename(previewfile), filebytes, preview_mimetype)},
-                                    verify=connector.ssl_verify if connector else True)
-        else:
-            result = connector.post(url, files={"File": filebytes}, verify=connector.ssl_verify if connector else True)
+    # upload visualization URL
+    visualization_url = '%s/api/v2/visualizations?name=%s&description=%s' % (
+        client.host, visualization_name, visualization_description)
```
Comment (longshuicy) [inline pyclowder/api/v2/files.py:178]: you might have to update that part to match the change in the endpoint

you'll have to swith the order you do things: 
- first register the config
- then post the visualization data with config id there

Comment (sandeep-ps) [reply pyclowder/api/v2/files.py]: Yep, I just did the above items. :-D

**Comment by sandeep-ps** [comment]
There is one build error, but I don't think it's related to this PR.

**Review by longshuicy — APPROVED** [review]
it working well with the frontend PR https://github.com/clowder-framework/clowder2/pull/583
approve.

**Review by ddey2 — APPROVED** [review]
Tested, works.

**Review by max-zilla — APPROVED** [review]

**Pull request merged**